include/nil/crypto3/zk/snark/systems/plonk/placeholder/preprocessor.hpp

```diff
@@ -453,6 +459,9 @@ namespace nil {
                         // TODO: add std::vector<std::size_t> columns_with_copy_constraints;
                         cycle_representation permutation(constraint_system, table_description);
 
+                        /* vo rewrite to vector of shared_ptr ?

```

Comment:
Please consider doing this as a part of this PR. We have some time now, if left we will probably never come back to do this.

Reply:
precommitments() takes const references, no need to rewrite it.
or at least I do not see the benefits of this proposal at my current level.
removed comment, updated in 528edcf5